src/main/java/de/tum/in/www1/artemis/service/util/ZonedDateTimeUtil.java

```diff
@@ -0,0 +1,29 @@
+package de.tum.in.www1.artemis.service.util;
+
+import java.time.ZonedDateTime;
+
+import jakarta.validation.constraints.NotNull;
+
+public class ZonedDateTimeUtil {
+
+    /**
+     * Private constructor to prevent instantiation.
+     */
+    private ZonedDateTimeUtil() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    /**
+     * Get the relative time of a ZonedDateTime object compared to an origin and a unit ZonedDateTime object in percent.
+     * <p>
+     * Example: origin = 0:00, unit = 10:00, target = 2:30 => 25%
```

Comment:
Maybe the documentation could better capture that target = 100. Because 25% technically means 0.25 and not 25